core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/localscan/LocalCsvFileScanOpDesc.java

```diff
@@ -86,17 +82,102 @@ public Schema sourceSchema() {
     }
 
     private Schema inferSchema(String headerLine) {
+        java.util.List<AttributeType> attributeType = IntStream
+                .range(0, headerLine.split(delimiter).length).mapToObj(i->AttributeType.INTEGER).collect(Collectors.toList());
+
+
+        try(BufferedReader reader = new BufferedReader(new FileReader(filePath))) {
+            String line;
+            if (header != null && header)
+                reader.readLine();
+            int i = 0;
+            while((line = reader.readLine())!=null && i<=100) {
+                String[] records = line.split((delimiter));
+                attributeType = inferline(attributeType, records);
+                i++;
+            }
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        java.util.List<AttributeType> finalAttributeType = attributeType;
+        System.out.println("a->"+finalAttributeType);
+//        System.out.println(Schema.newBuilder().add(IntStream.range(0, headerLine.split(delimiter).length).
+//                mapToObj(i -> new Attribute(l.get(i), finalAttributeType.get(i)))
+//                .collect(Collectors.toList())).build());
+
         if (delimiter == null) {
             return null;
         }
         if (header != null && header) {
-            return Schema.newBuilder().add(Arrays.stream(headerLine.split(delimiter)).map(c -> c.trim())
-                    .map(c -> new Attribute(c, AttributeType.STRING)).collect(Collectors.toList())).build();
+
+
+            return Schema.newBuilder().add(IntStream.range(0, headerLine.split(delimiter).length).
+                    mapToObj(i -> new Attribute(headerLine.split(delimiter)[i], finalAttributeType.get(i)))
+                    .collect(Collectors.toList())).build();
+//            return Schema.newBuilder().add(Arrays.stream(headerLine.split(delimiter)).map(c -> c.trim())
+//                    .map(c -> new Attribute(c, AttributeType.STRING)).collect(Collectors.toList())).build();
         } else {
             return Schema.newBuilder().add(IntStream.range(0, headerLine.split(delimiter).length).
-                    mapToObj(i -> new Attribute("column" + i, AttributeType.STRING))
+                    mapToObj(i -> new Attribute("column" + i, finalAttributeType.get(i)))
                     .collect(Collectors.toList())).build();
         }
     }
+    private java.util.List<AttributeType> inferline (java.util.List<AttributeType> attributeType, String[] line) {
```

Comment:
1. rename this method to camel case.
2. rename the arguments to match its type, for example `List<AttributeType> attributeTypes, String[] tokens`.

Reply:
1. Done.
2. Done